sentry_sdk/metrics.py

```diff
@@ -554,55 +613,87 @@ def _emit(
+    # type: (MetricTagsInternal) -> Dict[str, Any]
+    rv = {}  # type: Dict[str, Any]
+    for tag_name, tag_value in tags:
+        old_value = rv.get(tag_name)
+        if old_value is not None:
+            if isinstance(old_value, list):
+                old_value.append(tag_value)
+            else:
+                rv[tag_name] = [old_value, tag_value]
+        else:
+            rv[tag_name] = tag_value
+    return rv
 
 
 def _get_aggregator_and_update_tags(key, tags):
-    # type: (str, Optional[MetricTags]) -> Tuple[Optional[MetricsAggregator], Optional[MetricTags]]
+    # type: (str, Optional[MetricTags]) -> Tuple[Optional[MetricsAggregator], Optional[LocalAggregator], Optional[MetricTags]]
     """Returns the current metrics aggregator if there is one."""
     hub = sentry_sdk.Hub.current
     client = hub.client
     if client is None or client.metrics_aggregator is None:
-        return None, tags
+        return None, None, tags
+
+    experiments = client.options.get("_experiments", {})
 
     updated_tags = dict(tags or ())  # type: Dict[str, MetricTagValue]
     updated_tags.setdefault("release", client.options["release"])
     updated_tags.setdefault("environment", client.options["environment"])
 
     scope = hub.scope
+    local_aggregator = None
+
+    # We go with the low-level API here to access transaction information as
+    # this one is the same between just errors and errors + performance
     transaction_source = scope._transaction_info.get("source")
     if transaction_source in GOOD_TRANSACTION_SOURCES:
-        transaction = scope._transaction
-        if transaction:
-            updated_tags.setdefault("transaction", transaction)
+        transaction_name = scope._transaction
+        if transaction_name:
+            updated_tags.setdefault("transaction", transaction_name)
+        if scope._span is not None:
+            sample_rate = experiments.get("metrics_summary_sample_rate") or 0.0
+            should_summarize_metric_callback = experiments.get(
+                "should_summarize_metric"
+            )
+            if random.random() < sample_rate and (
+                should_summarize_metric_callback is None
+                or should_summarize_metric_callback(key, updated_tags)
+            ):
+                local_aggregator = scope._span._get_local_aggregator()
 
-    callback = client.options.get("_experiments", {}).get("before_emit_metric")
-    if callback is not None:
+    before_emit_callback = experiments.get("before_emit_metric")
+    if before_emit_callback is not None:
         with recursion_protection() as in_metrics:
             if not in_metrics:
-                if not callback(key, updated_tags):
-                    return None, updated_tags
+                if not before_emit_callback(key, updated_tags):
+                    return None, None, updated_tags
 
-    return client.metrics_aggregator, updated_tags
+    return client.metrics_aggregator, local_aggregator, updated_tags
```

Comment:
Looks like none of the call sites actually use `local_aggregator`, is that intentional?